include/xrpl/protocol/STIssue.h

```diff
@@ -67,6 +76,15 @@ class STIssue final : public STBase, CountedObject<STIssue>
     bool
     isDefault() const override;
 
+    friend constexpr std::weak_ordering
```

Comment:
You are getting infinite recursion in `STIssue::isEquivalent(const STBase& t)` because you also need to hide `bool STBase::operator==(const STBase& t) const` , and the way to do that is by adding:

```
friend constexpr bool
operator==(STIssue const& lhs, STIssue const& rhs)
{
    return lhs.asset_ == rhs.asset_;
}

```

Reply:
Thanks for figuring this out. I pushed a fix.